ui/ui-components/pages/app/[id]/edit.tsx

```diff
@@ -132,17 +206,17 @@ export default function Page(props) {
 
       <Row>
         <Col>
-          <Form id="form" onSubmit={edit} autoComplete="off">
+          <Form id="form" onSubmit={handleSubmit(edit)} autoComplete="off">
             <fieldset disabled={Boolean(app.locked)}>
               <Form.Group controlId="name">
                 <Form.Label>Name</Form.Label>
                 <Form.Control
                   required
                   type="text"
+                  name="name"
                   placeholder="Name"
-                  value={app.name}
                   disabled={loading}
-                  onChange={(e) => update(e)}
+                  ref={register}
```

Comment:
Moved to using `react-hook-form` on this page so that we could more easily update the fields as needed on incoming oauth responses.